webpack.config.js

```diff
@@ -0,0 +1,245 @@
+let ReplacePlugin = require('webpack-plugin-replace');
+let CopyPlugin = require('copy-webpack-plugin');
+let WatchTimestampsPlugin = require('./config/watch-timestamps-plugin');
+let BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+
+let oldWrite = process.stderr.write;
+process.stderr.write = chunk => {
+	if (String(chunk).indexOf('DeprecationWarning:')!==-1) return;
+	return oldWrite.call(process.stderr, chunk);
+};
+
+function parseJson(text, fallback) {
+	try {
+		return JSON.parse(text);
+	}
+	catch (e) { }
+	return fallback || {};
+}
+
+function readFile(filename) {
+	try {
+		return fs.readFileSync(filename);
+	}
+	catch (e) {}
+}
+
+module.exports = function(_, env) {
+	let isProd = env.mode === 'production';
+	let nodeModules = path.join(__dirname, 'node_modules');
+	let componentStyleDirs = [
+		path.join(__dirname, 'src/components'),
+		path.join(__dirname, 'src/routes')
+	];
+
+	let babelRc = parseJson(readFile('.babelrc'));
+	babelRc.babelrc = false;
+	if (isProd) babelRc.presets[0][1].modules = false;
+
+	let manifest = parseJson(readFile('./src/manifest.json'));
+
+	return {
+		mode: env.mode || 'development',
+		entry: path.join(__dirname, 'config/client-boot.js'),
+		// entry: './src/index',
+		output: {
+			filename: isProd ? '[name].[chunkhash:5].js' : '[name].js',
+			chunkFilename: '[name].chunk.[chunkhash:5].js',
+			path: path.join(__dirname, 'build'),
+			publicPath: '/'
+		},
+		resolve: {
+			extensions: ['.ts', '.tsx', '.js', '.jsx', '.scss', '.css'],
+			alias: {
+				'app-entry-point': path.join(__dirname, 'src/index'),
+				style: path.join(__dirname, 'src/style'),
+				preact$: path.join(__dirname, 'node_modules/preact/dist/preact.js')
+			}
+		},
+		resolveLoader: {
+			alias: {
+				async: path.join(__dirname, 'config/async-component-loader')
+			}
+		},
+		module: {
+			rules: [
+				{
+					test: /\.tsx?$/,
+					enforce: 'pre',
+					// loader: 'awesome-typescript-loader',
+					loader: 'ts-loader',
+					exclude: nodeModules
+				},
+				{
+					test: /\.(tsx?|jsx?)$/,
+					loader: 'babel-loader',
+					options: babelRc
+				},
+				{
+					test: /\.(scss|sass)$/,
+					loader: 'sass-loader',
+					enforce: 'pre',
+					options: {
+						sourceMap: true,
+						includePaths: [nodeModules]
+					}
+				},
+				{
+					test: /\.(scss|sass|css)$/,
+					include: componentStyleDirs,
+					use: [
+						isProd ? MiniCssExtractPlugin.loader : 'style-loader',
+						{
+							// loader: 'typings-for-css-modules-loader?modules&localIdentName=[local]__[hash:base64:5]&importLoaders=1'+(isProd ? '&sourceMap' : '')
+							// loader: 'css-loader',
+							loader: 'typings-for-css-modules-loader',
+							options: {
+								modules: true,
+								localIdentName: '[local]__[hash:base64:5]',
+								namedExport: true,
+								camelCase: true,
+								importLoaders: 1,
+								sourceMap: isProd,
+								sass: true
+							}
+						}
+					]
+					// loader: ExtractTextPlugin.extract({
```

Comment:
We have some of these commented blocks. We should either remove them or add a comment explaining how and when and why they can be re-enabled

Reply:
I'll remove them, they were from before I'd managed to get the new `mini-extract-plugin` working reliably.